compiler/damlc/tests/src/DA/Test/ScriptService.hs

```diff
@@ -158,6 +158,55 @@ main =
                     ]
                 expectScriptSuccess rs (vr "testExerciseByKey") $ \r ->
                   matchRegex r "Active contracts: \n\nReturn value: 42\n\n$",
+              testCase "fetch and exercising by key shows key in log" $ do
+                rs <-
+                  runScripts
+                    scriptService
+                    [ "module Test where",
+                      "import Daml.Script",
+                      "",
+                      "template T",
+                      "  with",
+                      "    owner : Party",
+                      "  where",
+                      "    signatory owner",
+                      "    key owner : Party",
+                      "    maintainer key",
+                      "    nonconsuming choice C : ()",
+                      "      controller owner",
+                      "      do",
+                      "        pure ()",
+                      "",
+                      "template Runner",
+                      "  with",
+                      "    owner : Party",
+                      "  where",
+                      "    signatory owner",
+                      "    choice Run : ()",
+                      "      with",
+                      "        party : Party",
+                      "      controller owner",
+                      "      do",
+                      "        cid <- create T with owner = party",
+                      "        exercise cid C",
+                      "        exerciseByKey @T party C",
+                      "        fetch cid",
+                      "        fetchByKey @T party",
+                      "        pure ()",
+                      "",
+                      "test = do",
+                      "  p <- allocateParty \"p\"",
+                      "  submit p $ createAndExerciseCmd (Runner p) (Run p)"
```

Comment:
can we add a fetch as well and for both of those test that we get the `using key` in the byKey case but not if it’s a regular fetch or exercise.